crates/core/src/protocols/evalcheck/verify.rs

```diff
@@ -253,6 +253,10 @@ impl<'a, F: TowerField> EvalcheckVerifier<'a, F> {
 					.evaluate(zs)
 					.expect("select_row is constructor with zs.len() variables");
 
+				if eval.is_zero() && select_row_term.is_zero() {
+					return Ok(());
+				}
+
 				let inner_eval = eval * select_row_term.invert_or_zero();
```

Comment:
You should be able to exit whenever `select_row_term.is_zero()`. If `eval` is zero then it's OK, if `eval` is nonzero, then it's an incorrect evaluation error.

You then shouldn't need `invert_or_zero()`, just `invert`